x-pack/plugins/lens/public/lens_suggestions_api.ts

```diff
@@ -86,9 +88,40 @@ export const suggestionsApi = ({
     dataViews,
   }).filter((sug) => !sug.hide && sug.visualizationId !== 'lnsLegacyMetric');
   const suggestionsList = [activeVisualization, ...newSuggestions];
+
+  // in case the user asks for a specific type, if it exists in the list, prioritize it
+  if (suggestionsList.length > 1 && preferredChartType) {
+    const suggestionFromModel = suggestionsList.find(
+      (s) => s.title.includes(preferredChartType) || s.visualizationId.includes(preferredChartType)
+    );
+    if (suggestionFromModel) {
+      return [suggestionFromModel];
+    } else {
+      // handle here the line, area logic
+      if (
```

Comment:
move this outside the if, should run every time there is an XY suggestion and the preferredType is line or area